src/games/ecards.py

```diff
@@ -148,24 +151,27 @@ async def execute_round(self, leader: int, prompt: str, hands: dict):
 
         # get replies
         replies: Dict[int, int] = view.responses
-        if TestBotUser.test_bot_id in self.players and TestBotUser.test_bot_id != leader:
-            replies[TestBotUser.test_bot_id] = random.choice(range(len(hands[TestBotUser.test_bot_id])))
+        if (
+            TestBotUser.test_bot_id in self.players
+            and TestBotUser.test_bot_id != leader
+        ):
+            replies[TestBotUser.test_bot_id] = random.choice(
+                range(len(hands[TestBotUser.test_bot_id]))
+            )
 
         # unpack which card played
         # pid -> str
         cards_played = {}
         for pid, responseIndex in replies.items():
 
             if responseIndex is None:
-                self.logging.info(
-                    f"Player {self.players[pid]} did not play a card."
-                )
+                self.logging.info(f"Player {self.players[pid]} did not play a card.")
             else:
                 cards_played[pid] = hands[pid].pop(responseIndex)
                 self.logging.info(f"player {pid} chose {responseIndex}")
 
         # shuffle responses to list
-        shuffled_responses = list(cards_played.values())
+        shuffled_responses = [(v, k) for (k, v) in cards_played.items()]
```

Comment:
Looking at this again it feels a bit arbitrary that we switch the ordering here. Make a mental note to go through the code and keep the order consistent.